src/Kit/System/EventLoop.h

```diff
@@ -59,11 +64,20 @@ class EventLoop : public IRunnable, public IEventManager, public ISignable, publ
         When the 'eventFlagsList' parameter is null then the EventLoop will not
         monitor any event flags.
 
+        The 'watchdog' parameter is optional and can be used to provide
+        watchdog monitoring for this event loop.
+
         WARNING: The application CANNOT modify the eventFlagsList after the
                  EventLoop instance is created.
      */
     EventLoop( uint32_t                           timeOutPeriodInMsec = OPTION_KIT_SYSTEM_EVENT_LOOP_TIMEOUT_PERIOD,
-               Kit::Container::SList<IEventFlag>* eventFlagsList      = nullptr ) noexcept;
+               Kit::Container::SList<IEventFlag>* eventFlagsList      = nullptr,
+#ifdef USE_KIT_SYSTEM_WATCHDOG
+               Kit::System::Watchdog::WatchedEventLoopApi* watchdog = nullptr
+#else
+               void* watchdog = nullptr  // Placeholder when watchdog is disabled
+#endif
+               ) noexcept;
```

Comment:
See coding standard about #ifdef usage.
```suggestion
               Kit::Container::SList<IEventFlag>* eventFlagsList      = nullptr,
               Kit::System::Watchdog::IWatchedEventLoop* watchdog = nullptr
 ) noexcept;
```